### PR TITLE
Header UI changes and Social Sharing of the App

### LINKED ISSUE #178: Social Sharing of the App
## Is your feature request related to a problem? Please describe.

We can share this mobile app with others via whatsapp ,  twitter , mail, etc.

## Describe the solution you'd like

Add social sharing with this application.

@KeenWarrior @Abhishek-kumar09 Can I work on this ?

### PR DESCRIPTION
**Pull request by poojadas07**
**Issue Number**

fixes #178  , #258 

**Describe the changes you've made**

I added a design for social sharing of the app and also changed the UI of the header for all pages.

**Checklist**

 - [x] My code follows the code style of this project.
 - [] My change requires a change to the documentation.
 - [] I have updated the documentation accordingly.
 - [x] All new and existing tests passed.
 - [x] The title of my pull request is a short description of the requested changes.

wishlist
courses
social
dashboard
form
profile

### CODE REVIEW

**Comment by poojadas07** [comment]
> _No description provided._

wishlist
courses
social
dashboard
form
profile

**Comment by sakshi-choudhary** [comment]
I think the social sharing has to work as well @poojadas07 i.e add functionality as well, you can refer capacitor docs for it.

**Comment by poojadas07** [comment]
> I think the social sharing has to work as well @poojadas07 i.e add functionality as well, you can refer capacitor docs for it.

Yeahh It is working.

**Comment by sakshi-choudhary** [comment]
> > I think the social sharing has to work as well @poojadas07 i.e add functionality as well, you can refer capacitor docs for it.
> 
> Yeahh It is working.

Okay, I'm not sure but I tried it on my phone and it didn't redirect me to whatsapp or linkedin. Is this compatible with ios as well?

**Comment by poojadas07** [comment]
@Abhishek-kumar09 Please check it once.

**Comment by Abhishek-kumar09** [comment]
Yeah the sharing option is working perfectly in android devices.
@sakshi-choudhary Do you have iOS, is it not opening whatsapp, and others?

File: src/pages/CoursePageViewWithVideo/index.js [hunk 1]
```diff
@@ -20,40 +26,60 @@ const useStyles = makeStyles((theme) => ({
...
+          </Typography>
+          <IconButton edge="end" className={classes.menuButton} color="inherit" aria-label="menu">
+            <MoreVertIcon />
+          </IconButton>
+        </Toolbar>
+      </AppBar>
+      <Container className={classes.root}>
+        <ScrollTop />
+        {/* <Typography variant="h2" align="center">
         Your Courses
```
Comment (Abhishek-kumar09) [inline src/pages/CoursePageViewWithVideo/index.js:62]: Remove This

File: src/pages/CoursePageViewWithVideo/index.js [hunk 2]
```diff
@@ -20,40 +26,60 @@ const useStyles = makeStyles((theme) => ({
...
-      </List>
-      <div className={classes.backtotop}>
-        <ScrollTop {...props}>
-          <Fab color="secondary" size="small" aria-label="scroll back to top">
-            <KeyboardArrowUpIcon />
-          </Fab>
-        </ScrollTop>
-      </div>
-    </div>
+      </Typography> */}
```
Comment (Abhishek-kumar09) [inline src/pages/CoursePageViewWithVideo/index.js:63]: remove this

File: src/pages/ProfilePage/ProfileComponent/ListComponent.js [hunk 3]
```diff
@@ -14,6 +14,10 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 export default function ListComponent() {
+  // const [checked, setChecked] = React.useState(false);
+  // const toggleChecked = () => {
+  //   setChecked((prev) => !prev);
+  // };
   const classes = useStyles();
```
Comment (Abhishek-kumar09) [inline src/pages/ProfilePage/ProfileComponent/ListComponent.js:21]: Please Revert it.

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

**Comment by sakshi-choudhary** [comment]
> Yeah the sharing option is working perfectly in android devices.
> @sakshi-choudhary Do you have iOS, is it not opening whatsapp, and others?

yes, I have ios, so for me whatsapp and Linkedin wasn't working. I'll try again and let know

File: src/pages/ProfilePage/ProfileComponent/ListComponent.js (hunk 3 above)
Comment (poojadas07) [reply src/pages/ProfilePage/ProfileComponent/ListComponent.js:21]: But we are not using these anywhere thats why it is showing error

File: src/pages/ProfilePage/ProfileComponent/HeaderComponent.js [hunk 4]
```diff
@@ -80,7 +124,49 @@ function HeaderComponent() {
...
+              onClose={handleClose}
+              closeAfterTransition
+              BackdropComponent={Backdrop}
+              BackdropProps={{
+                timeout: 500,
+              }}
+            >
+              <Fade in={open}>
+                <div className={classes.paper}>
+                  <h2 style={{ textAlign: "center" }}>Share with</h2>
```
Comment (sakshi-choudhary) [inline src/pages/ProfilePage/ProfileComponent/HeaderComponent.js:142]: You can use `typography` here @poojadas07 , this font style doesn't seem to go with the app theme.

**Comment by sakshi-choudhary** [comment]
> > Yeah the sharing option is working perfectly in android devices.
> > @sakshi-choudhary Do you have iOS, is it not opening whatsapp, and others?
> 
> yes, I have ios, so for me whatsapp and Linkedin wasn't working. I'll try again and let know

this seems to be working fine now.

File: src/pages/ProfilePage/ProfileComponent/HeaderComponent.js (hunk 4 above)
Comment (poojadas07) [reply src/pages/ProfilePage/ProfileComponent/HeaderComponent.js:142]: okay

**Pull request closed without merging**